blockdag/blocknode_test.go

```diff
@@ -95,3 +96,48 @@ func TestChainHeight(t *testing.T) {
 	}
 
 }
+
+func TestKTypeIsUnsigned(t *testing.T) {
+	k := dagconfig.KType(0)
+	k--
+
+	if k < dagconfig.KType(0) {
+		t.Fatalf("KType must be unsigned")
+	}
+}
+
+// This test is to ensure the size BlueAnticoneSizesSize is serialized to the size of KType.
+// We verify that by serializing and deserializing the block while we make sure that we stay within the expected range.
+func TestBlueAnticoneSizesSize(t *testing.T) {
+	dag, teardownFunc, err := DAGSetup("TestBlueAnticoneSizesSize", Config{
+		DAGParams: &dagconfig.SimnetParams,
+	})
+	if err != nil {
+		t.Fatalf("TestBlueAnticoneSizesSize: Failed to setup DAG instance: %s", err)
+	}
+	defer teardownFunc()
+	blockHeader := dagconfig.SimnetParams.GenesisBlock.Header
+	node, _ := dag.newBlockNode(&blockHeader, newSet())
+	hash := daghash.Hash{1}
+
+	// Setting maxKType to maximum value fo KType
```

Comment:
`^dagconfig.KType(0)` is maximum value of KType because we know for sure KType is unsigned (otherwise it would be -1 for signed types).
Please mention in the comment this assumption, and that it's true as long as TestKTypeIsUnsigned doesn't fail.

Reply:
Also: typo: fo -> of